huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go

```diff
@@ -0,0 +1,512 @@
+package dli
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/dli/v1/flinkjob"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func ResourceFlinkJarJob() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFlinkJarJobCreate,
+		ReadContext:   resourceFlinkJarJobRead,
+		UpdateContext: resourceFlinkJarJobUpdate,
+		DeleteContext: resourceFlinkJarJobDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 57),
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 512),
+			},
+
+			"queue_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"main_class": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"entrypoint": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"entrypoint_args": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"dependency_jars": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"dependency_files": {
```

Comment:
maybe `dependent_xxx` is better.

Reply:
keep it.  it standard in java maven project.